resolver/resolver.go

```diff
@@ -3320,18 +3354,54 @@ const privateProtoFile = "grpc/federation/private.proto"
 
 func messageArgumentFileDescriptor(arg *Message) *descriptorpb.FileDescriptorProto {
 	desc := arg.File.Desc
+	msg := messageArgumentDescriptor(arg)
+	var importedPrivateFile bool
+	for _, dep := range desc.GetDependency() {
+		if dep == privateProtoFile {
+			importedPrivateFile = true
+			break
+		}
+	}
+	deps := append(desc.GetDependency(), arg.File.Name)
+	if !importedPrivateFile {
+		deps = append(deps, privateProtoFile)
+	}
+	return &descriptorpb.FileDescriptorProto{
+		Name:             proto.String(arg.Name),
+		Package:          proto.String(federation.PrivatePackageName),
+		Dependency:       deps,
+		PublicDependency: desc.PublicDependency,
+		WeakDependency:   desc.WeakDependency,
+		MessageType:      []*descriptorpb.DescriptorProto{msg},
+	}
+}
+
+func messageArgumentDescriptor(arg *Message) *descriptorpb.DescriptorProto {
 	msg := &descriptorpb.DescriptorProto{
 		Name: proto.String(arg.Name),
+		Options: &descriptorpb.MessageOptions{
+			MapEntry: proto.Bool(arg.IsMapEntry),
+		},
 	}
 	for idx, field := range arg.Fields {
-		var typeName string
-		if field.Type.Message != nil {
+		var (
+			typeName string
+			label    = descriptorpb.FieldDescriptorProto_LABEL_OPTIONAL
+		)
+		if field.Type.Message != nil && field.Type.Message.IsMapEntry {
+			mp := messageArgumentDescriptor(field.Type.Message)
+			if mp.GetName() == "" {
```

Comment:
I didn't expect it to be empty, but under what circumstances would it be empty ?

Reply:
I understand ! OK 👍